examples/misc/tickedstroke_demo.py

```diff
@@ -0,0 +1,83 @@
+"""
+============================
+TickedStroke Patheffect Demo
+============================
+
+"""
+
+import matplotlib.patches as patches
+from matplotlib.path import Path
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.patheffects as PathEffects
+
+# Direct to path
+plt.figure(figsize=(6, 6))
+ax1 = plt.subplot(111)
+path = Path.unit_circle()
+patch = patches.PathPatch(path, facecolor='none', lw=2, path_effects=[
+    PathEffects.withTickedStroke(angle=90, spacing=10, length=1, flip=True)])
+
+ax1.add_patch(patch)
+ax1.axis('equal')
+ax1.set_xlim(-2, 2)
+ax1.set_ylim(-2, 2)
+
+plt.show()
+
+# Lines and curves with plot and legend
+plt.figure(figsize=(6, 6))
+ax3 = plt.subplot(111)
+p1, = ax3.plot([0, 1], [0, 1],
+               path_effects=[PathEffects.withTickedStroke(spacing=7)])
+
+nx = 101
+x = np.linspace(0.0, 1.0, nx)
+y = 0.3*np.sin(x*8) + 0.4
+p2, = ax3.plot(x, y, path_effects=[PathEffects.withTickedStroke()])
+
+leg = ax3.legend([p1, p2], ["Line", "Curve"], fancybox=True,
+                 loc='upper left', edgecolor='k')
+
+plt.show()
+
+# Contour plot with objective and constraints
+plt.figure(figsize=(6, 6))
+ax4 = plt.subplot(111)
+
+nx = 101
+ny = 105
+
+# Set up survey vectors
+xvec = np.linspace(0.001, 4.0, nx)
+yvec = np.linspace(0.001, 4.0, ny)
+
+# Set up survey matrices.  Design disk loading and gear ratio.
+x1, x2 = np.meshgrid(xvec, yvec)
+
+# Evaluate some stuff to plot
+obj = x1**2+x2**2-2*x1-2*x2+2
+g1 = -(3*x1+x2-5.5)
+g2 = -(x1+2*x2-4)
+g3 = .8+1/(x1**3)-x2
+
+cntr = plt.contour(x1, x2, obj, [0.01, 0.1, 0.5, 1, 2, 4, 8, 16],
+                   colors=('k',))
+ax4.clabel(cntr, fmt="%2.1f", use_clabeltext=True)
+
+cg1 = plt.contour(x1, x2, g1, [0], colors=('k',))
+plt.setp(cg1.collections,
+         path_effects=[PathEffects.withTickedStroke(angle=135)])
```

Comment:
Would it be reasonable to use `angle=45`? That way we would have
~~~
 / /     instead of:  \ \     
-----                  -----
~~~
Which fits better in the legend and also removes the impression that a tick is missing here:
![grafik](https://user-images.githubusercontent.com/2836374/67632622-13aee780-f8a6-11e9-8022-6824cc425720.png)

Reply:
You've grabbed the wrong one here -- the legend is only in the second example right now (I had trouble getting it to work with all the contours in the last example).

However, I've tried switching the one with the example from 45 to 135 to see if that does what you want.

You pretty much always end up with an odd start/end somewhere with these, not really worth fighting it IMO.

Perhaps changing :

```
num = int(np.ceil(stot / spcpx))
# Pick parameter values for ticks.
s_tick = np.linspace(0, stot, num)
```

to

```
num = int(np.ceil(stot / spcpx))-1
# Pick parameter values for ticks.
s_tick = np.linspace(spcpx/2, stot-spcpx/2, num)
```

Would result in a better effect?  I'll give it a shot...

Reply:
The half offset looks really good in the [examples](https://26079-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/misc/tickedstroke_demo.html#sphx-glr-gallery-misc-tickedstroke-demo-py).